### PR TITLE
Remove Mat/Vec template parameters #2

### PR DESCRIPTION
**Pull request by norihiro-w**
follow up of #1257

### CODE REVIEW

File: ProcessLib/TES/TESProcess.cpp [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 #include "TESProcess.h"
 
+#include "Utils/CreateLocalAssemblers.h"
```
Comment (chleh) [inline ProcessLib/TES/TESProcess.cpp]: Sall we use a full path here, i.e. prepend `ProcessLib`?

File: NumLib/ODESolver/MatrixTranslator.h [hunk 2]
```diff
@@ -25,85 +25,80 @@ namespace NumLib
 
 
 /*! Translates matrices and vectors assembled by a provided ODE (or other equation)
- *  to the stiffness matrix and right-hand side vector of a linear equation system
+ *  to the stiffness GlobalMatrix and right-hand side vector of a linear equation system
```
Comment (chleh) [inline NumLib/ODESolver/MatrixTranslator.h]: I'd keep the old comment.

File: MathLib/LinAlg/LinearSolver.h [hunk 3]
```diff
@@ -20,50 +20,15 @@ class ConfigTree;
 namespace MathLib
 {
 
-//! Basic linear solver interface.
-template<typename Matrix, typename Vector>
-class LinearSolver
```
Comment (chleh) [inline MathLib/LinAlg/LinearSolver.h]: If you kept the linear solver base class, we could in principle have EigenLinearSolver and EigenLISLinearSolver (and later on maybe others, too) in a single build.

**Comment by chleh** [comment]
Nice :+1:. Maybe now even the compile time reduces a bit?
Now some lines in NumLib/ODESolver are even longer than before. Could you clang-format everything in NumLib/ODESolver, please? I hope that won't change things (i.e. doc comments) to the worse.

**Comment by endJunction** [comment]
There is a clang-format-diff python script for application of clang-format to the diffs only.

File: MathLib/LinAlg/LinearSolver.h (hunk 3 above)
Comment (norihiro-w) [inline MathLib/LinAlg/LinearSolver.h]: we can discuss it when such a feature is required

**Comment by norihiro-w** [comment]
formatted

File: NumLib/DOF/GlobalMatrixProviders.h [hunk 4]
```diff
@@ -14,17 +14,14 @@
 
 namespace MathLib
 {
-
-template<typename Vector>
 struct GlobalVectorProvider
 {
-    static VectorProvider<Vector>& provider;
+    static VectorProvider<GlobalVector>& provider;
```
Comment (endJunction) [inline NumLib/DOF/GlobalMatrixProviders.h]: Not following all the ideas here, just a question: Why is GlobalVector template argument needed? Couldn't the VectorProvider just use the global type of vectors?

File: NumLib/ODESolver/MatrixTranslator.h [hunk 5]
```diff
@@ -12,374 +12,249 @@
...
- * \tparam Vector the type of the solution vector of the equation
+/*! Translates matrices assembled by a provided first order implicit
+ * quasi-linear ODE to some other matrices suitable to be passed on to nonlinear solvers.
  *
  * \see ODESystemTag::FirstOrderImplicitQuasilinear
  */
-template<typename Matrix, typename Vector>
-class MatrixTranslator<Matrix, Vector, ODESystemTag::FirstOrderImplicitQuasilinear>
+template <>
+class MatrixTranslator<ODESystemTag::FirstOrderImplicitQuasilinear>
```
Comment (endJunction) [inline NumLib/ODESolver/MatrixTranslator.h]: I think the specializations can also go into the cpp file, but I might be wrong about this.

**Comment by endJunction** [comment]
:fast_forward: so far.
I'd have split this PR in several parts, because reviewing +-2kloc changes is difficult and error prone. I definitely didn't get all the changes and some changes are "mechanical" (like sed scripts) other changes are by hand and need extra attention (which they didn't get from my side).

File: NumLib/ODESolver/MatrixTranslator.h (hunk 5 above)
Comment (chleh) [inline NumLib/ODESolver/MatrixTranslator.h]: No, the specializations are entirely different types. So the declarations of the members must stay here.

File: NumLib/DOF/GlobalMatrixProviders.h (hunk 4 above)
Comment (chleh) [inline NumLib/DOF/GlobalMatrixProviders.h]: Should be possible since the Eigen::MatrixXd cases are removed from TestODEInt.cpp

**Comment by norihiro-w** [comment]
@endJunction Have you made review commit by commit? Basically I deleted the template parameters and moved some implementations to cpp files. Large diffs are caused by clang-format.

**Comment by endJunction** [comment]
In single chunk.

**Comment by norihiro-w** [comment]
i will try restructuring commits for helping reviews. let me know if you have suggestions

**Comment by norihiro-w** [comment]
changed commits to include removing the template parameters in NumLib/ODESolver dir and resulting changes in other classes. see commits titled "rm Mat/Vec templates from xxxx".

**Comment by norihiro-w** [comment]
Jenkins, test this please!

**Comment by norihiro-w** [comment]
build time: `make -j4` for cli, test, dynamic link, no lis
2m52.551s (before this PR) vs 2m50.176s (with this PR)

File: NumLib/ODESolver/MatrixTranslator.h (hunk 5 above)
Comment (norihiro-w) [inline NumLib/ODESolver/MatrixTranslator.h]: This will be done in #1283

File: NumLib/DOF/GlobalMatrixProviders.h (hunk 4 above)
Comment (norihiro-w) [inline NumLib/DOF/GlobalMatrixProviders.h]: This will be done in #1283

**Comment by endJunction** [comment]
@norihiro-w Would you please rebase it?

**Comment by norihiro-w** [comment]
rebased

File: NumLib/ODESolver/EquationSystem.h [hunk 6]
```diff
@@ -30,9 +29,7 @@ enum class IterationResult : char
  *
  * \tparam Vector the type of the solution vector of the equation.
```
Comment (endJunction) [inline NumLib/ODESolver/EquationSystem.h]: I didn't see it before, but all of the Doxygen documentation needs a fixup; the template parameter does not exist anymore.

Comment (norihiro-w) [inline NumLib/ODESolver/EquationSystem.h]: i will fix them

**Pull request merged**